exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/managed/PriorityQueueCopierTemplate.java

```diff
@@ -57,9 +57,12 @@ public void setup(FragmentContext context, BufferAllocator allocator, VectorAcce
 
     queueSize = 0;
     for (int i = 0; i < size; i++) {
-      vector4.set(i, i, batchGroups.get(i).getNextIndex());
-      siftUp();
-      queueSize++;
+      int index = batchGroups.get(i).getNextIndex();
+      vector4.set(i, i, index);
```

Comment:
If the record batch is empty, why set this (65535) value in the vector ?
Probably usually the empty batch is last, so queueSize would not be incremented; otherwise if the next iteration brings a non empty batch, then siftUp() may run into that 65535 .... (not sure ...)